async_processor/types.py

```diff
@@ -74,6 +74,11 @@ class KafkaSASLAuth(BaseModel):
     password: str
 
 
+class NATSUserPasswordAuth(BaseModel):
```

Comment:
NATSBasicAuth?

Reply:
They do not call it basic auth.

Basic Auth is an HTTP concept; NATS do not use HTTP.
https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/username_password#plain-text-passwords